staff/sys/ldap-lint

```diff
@@ -66,12 +66,12 @@ def main():
         )
         for attrs in map(itemgetter('attributes'), c.response):
             cn = attrs['cn'][0]
-            type_ = attrs['type'][0]
+            type_ = attrs['type']
 
             if type_ not in RECOGNIZED_TYPES:
                 complain(cn, 'has unknown type ' + type_)
 
-            if 'macAddress' in attrs:
+            if attrs.get('macaddress', []):
```

Comment:
The capitalization is a typo, thanks for spotting. I tested it before refactoring and making the typo, though.

LDAP entries can have multiple `macAddress` attributes, so `attrs['macAddress']` is supposed to be a list. When an entry doesn't have a `macAddress` entry, `attrs['macAddress']` is an empty list. In the old version, `attrs` simply didn't have the key `'macAddress'`.

`ldap-lint` should probably be updated to complain if multiple `macAddress` attributes are defined. (Unless the schema supports specifying only zero or one of an attribute?)